controllers/vspherecluster_controller.go

```diff
@@ -240,6 +244,26 @@ func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconci
 	}
 	conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
 
+	// Remove finalizer on Identity Secret
+	if identity.IsSecretIdentity(ctx.VSphereCluster) {
+		secret := &apiv1.Secret{}
+		secretKey := client.ObjectKey{
+			Namespace: ctx.VSphereCluster.Namespace,
+			Name:      ctx.VSphereCluster.Spec.IdentityRef.Name,
+		}
+		err := ctx.Client.Get(ctx, secretKey, secret)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		if !apierrors.IsNotFound(err) {
+			r.Logger.Info(fmt.Sprintf("Removing finalizer form Secret %s/%s", secret.Namespace, secret.Name))
+			ctrlutil.RemoveFinalizer(secret, infrav1.SecretIdentitySetFinalizer)
```

Comment:
do we really need a finalizer ? i.e. are there things we want to perform before deleting the secret ?

Reply:
spoke about the finalizer offline. just to prevent accidental deletion of a secret that's still in use by a cluster.